File: dart/lib/src/v1/models/api_request.dart
import 'message_model.dart';
import 'user_data.dart';
import 'wallet_type.dart';

class ApiRequest {
  String method;
  UserData userdata;
  List<MessageModel>? messageData;
  WalletType? walletType;

  ApiRequest({
    required this.method,
    required this.userdata,
    this.messageData,
    this.walletType,
  });

  Map<String, dynamic> toJson() => {
    'method': method,
    'userdata': userdata.toJson(),
    'msgdata': messageData?.map((e) => e.toJson()).toList(),
    'walletType': walletType?.value,
  };
}

File: kotlin/src/main/kotlin/v1/models/ApiRequest.kt
package v1.models

import com.fasterxml.jackson.annotation.JsonProperty

class ApiRequest {
    @JsonProperty(required = true)
    var method: String? = null

    @JsonProperty(required = true)
    var userdata: UserData? = null

    @JsonProperty(value = "msgdata")
    var messageData: MutableList<MessageModel>? = null
}
